samples/java_layout/src/main/java/io/customer/android/sample/java_layout/ui/user/AuthViewModel.java

```diff
@@ -0,0 +1,49 @@
+package io.customer.android.sample.java_layout.ui.user;
+
+import android.os.Build;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.LiveDataReactiveStreams;
+import androidx.lifecycle.ViewModel;
+
+import io.customer.android.sample.java_layout.data.PreferencesDataStore;
+import io.customer.android.sample.java_layout.data.model.User;
+import io.customer.android.sample.java_layout.support.Optional;
+import io.reactivex.rxjava3.core.Flowable;
+
+public class AuthViewModel extends ViewModel {
+    private final PreferencesDataStore preferencesDataStore;
+    private final LiveData<User> userDataObservable;
+    private final LiveData<Boolean> userLoggedInStateObservable;
+
+    public AuthViewModel(PreferencesDataStore preferencesDataStore) {
+        this.preferencesDataStore = preferencesDataStore;
+        Flowable<Optional<User>> publisher = preferencesDataStore.userData().map(
+                bundle -> Optional.ofNullable(User.fromMap(bundle)));
+        this.userDataObservable = LiveDataReactiveStreams.fromPublisher(
+                publisher.filter(Optional::isPresent).map(Optional::get));
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            this.userLoggedInStateObservable = LiveDataReactiveStreams.fromPublisher(
+                    publisher.map(Optional::isPresent));
+        } else {
+            //noinspection Convert2MethodRef
+            this.userLoggedInStateObservable = LiveDataReactiveStreams.fromPublisher(publisher.map(optional -> optional.isPresent()));
+        }
```

Comment:
```suggestion
 public AuthViewModel(PreferencesDataStore preferencesDataStore) {
        this.preferencesDataStore = preferencesDataStore;
        Flowable<Optional<User>> publisher = preferencesDataStore.userData()
                .map(User::fromMap);
        this.userDataObservable = LiveDataReactiveStreams.fromPublisher(
                publisher.filter(Optional::isPresent).map(Optional::get));
        this.userLoggedInStateObservable = Transformations.map(userDataObservable, user -> user != null);
    }
```

We can probably use `Transformations` ?

Reply:
Had to tweak the suggestion slightly, updated code